app/schema/dogs/get.js

```diff
@@ -3,7 +3,9 @@ const Joi = require('joi')
 const dogsQueryParamsSchema = Joi.object({
   forPurging: Joi.boolean().optional(),
   sortKey: Joi.string().optional(),
-  sortOrder: Joi.string().optional().allow('ASC', 'DESC')
+  sortOrder: Joi.string().optional().allow('ASC', 'DESC'),
+  stepNum: Joi.number().optional().allow(1, 2),
```

Comment:
sorry to be a pain, but my solution architecture hat is coming on with this.  `stepNum` would argue is a f/e concern, as such this seems to be a backend for front end pattern (https://samnewman.io/patterns/architectural/bff & https://www.craigpardey.com/post/2022-11-28-bff). As far as I know this is not the direction we've gone down on this project.